### PR TITLE
fix(ast) Use AST in discover detect_table to better detect table

### PR DESCRIPTION
**Pull request by evanh**
Now that the AST is stable we can use it to find columns that the regular parser
doesn't expose, in particular the columns used in functions such as apdex.

### CODE REVIEW

File: tests/datasets/test_discover.py [hunk 1]
```diff
@@ -31,18 +31,22 @@ def get_dataset_source(dataset_name):
     ({"selected_columns": ["group_id"]}, "sentry_local"),
     ({"selected_columns": ["trace_id"]}, "transactions_local"),
     ({"selected_columns": ["group_id", "trace_id"]}, "sentry_local"),
-    ({"aggregations": [["max", "duration", "max_duration"]]}, "transactions_local",),
+    ({"aggregations": [["max", "duration", "max_duration"]]}, "transactions_local"),
+    (
+        {"aggregations": [["apdex(duration, 300)", None, "apdex_duration_300"]]},
+        "transactions_local",
```
Comment (fpacifici) [inline tests/datasets/test_discover.py:37]: nice

**Review by fpacifici — APPROVED** [review]
thanks

File: tests/datasets/test_discover.py (hunk 1 above)
Comment (tkaemming) [reply tests/datasets/test_discover.py:37]: ☝️  This is super cool

**Review by markstory — COMMENTED** [review]
:clap: :clap:

**Pull request merged**